### ISSUE TITLE
Authenticate endpoints return HTML messages on fail

### ISSUE THREAD

**Issue by bartversluijs** [balena-io/balena-sdk#1045]
## Expected Behavior
When checking if Node.js application is still authenticated with openBalena server using `isLoggedIn`, I expect it to return a boolean with `true / false`, even if current token is expired.

## Actual Behavior
When `true`, everything goes well. If never tried to login (no `~/.balena/token` file), it returns `false`. But if token is expired, it returns a 404 error trying to get `whoami` and I expected it to be `false` at that time.
```
BalenaRequestError: Request error: <!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /whoami</pre>
</body>
</html>

    at Object.<anonymous> (~/node_modules/balena-request/build/request.js:190:27)
    at Generator.next (<anonymous>)
    at fulfilled (~/node_modules/tslib/tslib.js:111:62)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  body: '<!DOCTYPE html>\n' +
    '<html lang="en">\n' +
    '<head>\n' +
    '<meta charset="utf-8">\n' +
    '<title>Error</title>\n' +
    '</head>\n' +
    '<body>\n' +
    '<pre>Cannot GET /whoami</pre>\n' +
    '</body>\n' +
    '</html>\n',
  statusCode: 404,
  requestOptions: {
    url: '/whoami',
    baseUrl: 'https://api.<balena-server>/',
    refreshToken: false,
    timeout: 59000
  }
}
```

## Steps to Reproduce the Problem

1. Run a openBalena server (Can't test it with balenaCloud atm)
2. Authenticate with server using `sdk.auth.login`
3. Wait till token expires (I don't really know the expiry time)
4. Check if SDK is logged in using `sdk.auth.isLoggedIn`

## Specifications

- SDK version: 15.24.0
- Browser: 
- Node version: 14.3.0
- Npm version: 6.14.5
- OS: macOS 11.1

## References

Internal discussion thread:
See: https://www.flowdock.com/app/rulemotion/p-opensource/threads/qnGgwmvppSbFItlU5_U9Rk5JtPn

**Comment by thgreasi** [comment]
Hi @bartversluijs ,
Thanks for reporting this.

After some digging I figured out that this happens because balena-request it trying to auto-refresh the token when your token is not fresh any more
See: https://github.com/balena-io-modules/balena-request/blob/1e1f0269772680da4ec6c02f58defcbcf34f4468/lib/request.js#L80-L89

and it does so by using the `/whoami` endpoint which at the moment is not part of open balena
See: https://github.com/balena-io-modules/balena-request/blob/1e1f0269772680da4ec6c02f58defcbcf34f4468/lib/request.js#L389

I will have to start an internal discussion about this before we can fix the issue.

In the meantime as a way to unblock you I would like to suggest you to use the `models.apiKey.create()` method o generate an api key which isn't expiring and use that to authenticate with the SDK using the `auth.loginWithToken()` method. When using an api key you should no longer face this 404 error.

**Comment by thgreasi** [comment]
After discussing this internally we ended up that open-balena should offer the `/user/v1/refresh-token` endpoint.
Once that's done we can fix this by just switching balena-request to use the new endpoint.
Depends-on: https://github.com/balena-io/open-balena-api/issues/575

**Comment by thgreasi** [comment]
The /user/v1/refresh-token endpoints are now available in open-balena-api v0.181.0 and balena-request has already been changed to use the new endpoitns in v11.3.0, which the SDK has picked up in v15.30.0
See: https://github.com/balena-io-modules/balena-request/blob/master/CHANGELOG.md#v1130
See: https://github.com/balena-io/balena-sdk/blob/master/CHANGELOG.md#15300---2021-03-12
